packages/blockchain-wallet-v4-frontend/src/scenes/Home/Banners/RewardsBanner/index.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { PureComponent } from 'react'
+import { FormattedMessage } from 'react-intl'
+import { connect, ConnectedProps } from 'react-redux'
+import { LinkContainer } from 'react-router-bootstrap'
+import { bindActionCreators } from '@reduxjs/toolkit'
+import styled from 'styled-components'
+
+import { Icon, Text } from 'blockchain-info-components'
+import { actions } from 'data'
+import { Analytics } from 'data/types'
+import { media } from 'services/styles'
+
+import { BannerButton, IconWrapper } from '../styles'
+
+const Wrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  border: 1px solid ${(props) => props.theme.grey000};
+  border-radius: 8px;
+  padding: 20px;
+
+  ${media.atLeastLaptop`
+    height: 80px;
+    padding: 0 20px;
+  `}
+  ${media.mobile`
+    padding: 12px;
+    flex-direction: column;
+  `}
+`
+const Row = styled.div`
+  display: flex;
+  align-items: center;
+`
+const Column = styled.div`
+  display: flex;
+  flex-direction: column;
+
+  & > div:first-child {
+    margin-bottom: 4px;
+  }
+`
+const PendingIconWrapper = styled(IconWrapper)`
+  background-color: ${(props) => props.theme.blue100};
+`
+const Copy = styled(Text)`
+  display: flex;
+  align-items: center;
+  margin-right: 20px;
+  font-size: 12px;
+`
+
+class StartEarningRewards extends PureComponent<Props> {
```

Comment:
why use old convention for components definition?

Reply:
please use functional components and hooks